base_api/tests/test_base.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2019 Ivan Yelizariev <https://it-projects.info/team/yelizariev>
+# Copyright 2019 Anvar Kildebekov <https://it-projects.info/team/fedoranvar>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+from odoo.tests.common import TransactionCase
+from odoo.tests import tagged
+
+
+@tagged('post_install', 'at_install')
+class TestBase(TransactionCase):
+
+    def test_search_or_create(self):
+        # define test variables
+        partner_obj = self.env['res.partner']
+        company_obj = self.env['res.company']
+        t_name = 'test_search_or_create'
+        #
+        # Test #1: Record creation
+        #
+        t_vals = {'name': t_name}
+        partner_obj.search([('name', '=', t_name)]).unlink()
```

Comment:
You delete records here, but don't it with next tests

            'name': 'TestParent',


Why?